Amend architecture docs (#521)
itchysats/itchysats: docs/ARCHITECTURE.md

Applying state changes locally first allows us to record the user's intention and provide instant (UI) feedback that we are working on making it happen.
Even if the user restarts the entire application, we can pick up where we left of and finish what we were meant to be doing.

### Testing the actor system

A downside of an actor system is that it is composed at runtime rather than compile-time.
In other words, it is fairly easy to write code that compiles and sends messages to actors but does not actually work at runtime because the required actor is not alive.
Thus, the primary risk we need to eliminate with our tests is the wiring of our actors, i.e. do all the individual actors send the correct messages at the right point in time?
We deliberately write these tests against the actor system.

> Any component not tested as part of this (like projections for the frontend) either need to be so simple that they don't require testing or should be moved into the actor system.

Reviewer: ```suggestion
Any component not tested as part of this (like projections for the frontend) either need to be so simple that they don't require testing or should be represented as an actor in the actor system.
```
